sycl/include/CL/sycl/INTEL/esimd/esimd_memory.hpp

```diff
@@ -158,14 +158,15 @@ scatter(T *p, simd<T, n * ElemsPerAddr> vals, simd<uint32_t, n> offsets,
                             pred.data());
 }
 
-// TODO @rolandschulz
-// Should follow existing std::simd naming for similar APIs - "copy_from" and
-// "copy_to" to avoid confusion.
-//
 /// Flat-address block-load.
 /// \ingroup sycl_esimd
+// TODO normally, this function should just delegate to
+// simd::copy_from for the deprecation period, but separate implementations are
+// needed for now, as simd::copy_from does not support cache hints yet.
+// This API, even though deprecated, can't be removed until then.
 template <typename T, int n, CacheHint L1H = CacheHint::None,
           CacheHint L3H = CacheHint::None>
+__SYCL_DEPRECATED("block_load is deprecated, use simd::copy_from.")
```

Comment:
@romanovvlad, I need to take this back. XXX is deprecated, use XYZ instead leads to unneeded duplication in the warning message:

```
D:\BuildBot\Product_worker_intel\sycl-win-x64-pr\llvm.src\sycl\test\esimd\odr.cpp:38:30: warning: 'block_load<int, 8, sycl::INTEL::gpu::CacheHint::None, sycl::INTEL::gpu::CacheHint::None>' is deprecated: block_load is deprecated, use simd::copy_from. [-Wdeprecated-declarations]
1 warning and 16 errors generated.
```